### PR TITLE
T4680 Fixed telegraf prometheus-client listen-address invalid format

### PR DESCRIPTION
**Pull request by ServerForge**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

https://phabricator.vyos.net/T4680

## Component(s) name

telegraf

## Proposed changes

There are two current issues with the telegraf prometheus-client listen-address configuration. It seems to accept multiple address values, However it seems to only be able to accept one address in the configuration file, It also does not provide brackets around IPv6 addresses causing a too many colon error when using a single IPv6 Address.

## How to test

Changes were testested by building vyos-1x package with the changes and then building a vyos image with the modified 1x package. 

Image with changes can be found at https://repo.serverforge.org/vyos/dev/sagitta/vyos-1.4-rolling-202209080518-amd64.iso

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: data/templates/telegraf/telegraf.j2 [hunk 1]
```diff
@@ -45,7 +45,11 @@
 ### Prometheus ###
 [[outputs.prometheus_client]]
   ## Address to listen on
+{%     if prometheus_client.listen_address | is_ipv6 %}
+  listen = "[{{ prometheus_client.listen_address if prometheus_client.listen_address is vyos_defined else '' }}]:{{ prometheus_client.port }}"
```
Comment (sever-sever) [inline data/templates/telegraf/telegraf.j2]: We have “bracketize_ipv6” function
For example 
https://github.com/vyos/vyos-1x/blob/627cfc6d6733998a721ac97f9f5bc44c4c2c6797/data/templates/snmp/etc.snmpd.conf.j2#L48

**Review by sever-sever — CHANGES_REQUESTED** [review]

**Comment by c-po** [comment]
Please also adjust your commit headline to include a valid task ID https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md

File: data/templates/telegraf/telegraf.j2 [hunk 2]
```diff
@@ -45,7 +45,11 @@
 ### Prometheus ###
 [[outputs.prometheus_client]]
   ## Address to listen on
+{%     if prometheus_client.listen_address | is_ipv6 %}
+  listen = "{{ prometheus_client.listen_address | bracketize_ipv6 }}:{{ prometheus_client.port }}"
+{%     else %}
   listen = "{{ prometheus_client.listen_address if prometheus_client.listen_address is vyos_defined else '' }}:{{ prometheus_client.port }}"
+{%     endif %}
```
Comment (sever-sever) [inline data/templates/telegraf/telegraf.j2]: I guess it should be 
```
listen = "{{ prometheus_client.listen_address | join() | bracketize_ipv6 if prometheus_client.listen_address is vyos_defined else '' }}:{{ prometheus_client.port }}
```

Comment (ServerForge) [reply data/templates/telegraf/telegraf.j2]: That is much cleaner, il try and update and test it later this evening.

**Review by sever-sever — APPROVED** [review]

File: data/templates/telegraf/telegraf.j2 [hunk 3]
```diff
@@ -45,7 +45,7 @@
 ### Prometheus ###
 [[outputs.prometheus_client]]
   ## Address to listen on
-  listen = "{{ prometheus_client.listen_address if prometheus_client.listen_address is vyos_defined else '' }}:{{ prometheus_client.port }}"
```
Comment (c-po) [inline data/templates/telegraf/telegraf.j2:48]: Why do we need join() if it‘s a single address?

Comment (ServerForge) [reply data/templates/telegraf/telegraf.j2:48]: It appears to work fine without the join, it has been removed.

**Pull request merged**